nix/pkgs/ms-buildenv/default.nix

```diff
@@ -0,0 +1,201 @@
+
+       # minimal "convenience" things as its common to use /bin/bash in containers
+       # this is not strictly necessary though as the PATH will resolve it anyway
+
+       mkdir -p bin usr/bin
+       ln -s /nix/var/nix/profiles/default/bin/sh bin/sh
+       ln -s /nix/var/nix/profiles/default/bin/env usr/bin/env
+       ln -s /nix/var/nix/profiles/default/bin/bash bin/bash
+
+      # setup shadow, bashrc
+       # instead of cat EOF magic, simply copy over some files to /etc
+       cp -r ${./root/etc} etc
+
+       chmod +w etc etc/group etc/passwd etc/shadow
+       # make sure /tmp exists which is used by cargo
+       mkdir -m 0777 tmp
+       # used for our mayastor RPC socket
+       mkdir -p -m 0777 var/tmp
+
+       # allow ubuntu ELF binaries to run. VSCode copies it's own.
+       mkdir -p lib64
+       ln -s ${stdenv.glibc}/lib64/ld-linux-x86-64.so.2 lib64/ld-linux-x86-64.so.2
+
+       # VSCode assumes that /sbin/ip exists, it phones home i guess
+       mkdir sbin
+       ln -s /nix/var/nix/profiles/default/bin/ip sbin/ip
+
+       # you must still call nix-channel update if you wish to install something
+       # we don't do that by default because we we are already pretty fat. The
+       # root user is assumed to be the default user here, so create a dir.
+
+       mkdir root
+       echo 'https://nixos.org/channels/nixpkgs-unstable' > root/.nix-channels
+    '';
+
+    config = {
+      Cmd = [ "/nix/var/nix/profiles/default/bin/bash" ];
+      Env = [
+        # set some environment variables so that cargo can find them
+        "PROTOC_INCLUDE=${protobuf}/include"
+        "LIBCLANG_PATH=${llvmPackages.libclang}/lib"
+        "LOCAL_ACRHIVE=${glibc}/lib/locale/locale-archive"
+
+        # C_INCLUDE_PATH is searched by compilers next to any includes it gets
+        # passed during invocation
+
+        "C_INCLUDE_PATH=${libspdk}/include/spdk:/nix/var/nix/profiles/default/include"
+
+        # compilers (like rustc) need to know where the libraries are
+        "LIBRARY_PATH=/nix/var/nix/profiles/default/lib"
+        # same for runtime but only needed for non nix binaries who are not elf patched
+        "LD_LIBRARY_PATH=/nix/var/nix/profiles/default/lib"
+        # ensure that the generated profile is loaded when we enter the container
+        "ENV=/nix/var/nix/profiles/default/etc/profile.d/nix.sh"
+
+        "PAGER=less"
+        "PATH=/nix/var/nix/profiles/default/bin"
+
+        # in order for tools that download stuff to work we need certs
+        "SSL_CERT_FILE=/nix/var/nix/profiles/default/etc/ssl/certs/ca-bundle.crt"
+        "GIT_SSL_CAINFO=/nix/var/nix/profiles/default/etc/ssl/certs/ca-bundle.crt"
+        "RUST_BACKTRACE=1"
+      ];
+      Labels = {
+        # https://github.com/microscaling/microscaling/blob/55a2d7b91ce7513e07f8b1fd91bbed8df59aed5a/Dockerfile#L22-L33
+        "org.label-schema.vcs-ref" = "master";
+        "org.label-schema.vcs-url" = "https://github.com/gila/mayastor";
```

Comment:
I followed the link and after reading the docker file there it's not more clear to me what these definitions are good for. And I think certainly there should not be a URL to your private mayastor repo. Maybe we can just remove those definitions?